tests/parsenum/main.c

```diff
@@ -0,0 +1,97 @@
+#include <stdint.h>
+#include <stdio.h>
+
+#include "parsenum.h"
+#include "warnp.h"
+
+#define PARSENUM_CMP(x, y)						\
+	((isnan((double)x) && isnan((double)y)) ||			\
+	    (isinf((double)x) == isinf((double)y) && isinf((double)x)) || \
```

Comment:
This is a synonym for isinf(x) && isinf(y), I think?

Also, doesn't this make PARSENUM_CMP(INFINITY, -INFINITY) return nonzero?